src/components/popup/popup.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import {Portal} from '../../../src';
+
+export type VerticalPosition =  'top' | 'center' | 'bottom';
+export type HorizontalPosition = 'left' | 'center' | 'right';
+
+export interface PositionPoint {
+    vertical: VerticalPosition;
+    horizontal: HorizontalPosition;
+}
+
+export interface PopupProps {
+    anchor: Element;
+    open?: boolean;
+    anchorPosition?: PositionPoint;
+    popupPosition?: PositionPoint;
+    syncWidth?: boolean;
+    maxHeight?: number;
+}
+
+export class Popup extends React.Component<PopupProps, {}> {
+    public static defaultProps: Partial<PopupProps> = {
+        open: false,
+        anchorPosition: {vertical: 'bottom', horizontal: 'left'},
+        popupPosition: {vertical: 'top', horizontal: 'left'},
+        syncWidth: true,
+        maxHeight: 500
+    };
+
+    public render() {
+        if (this.props.anchor && this.props.open) {
```

Comment:
Please add test and `| null` typing for the `this.props.anchor` as null (valid use-case, should be tested).